### DISCUSSION TITLE
isosurface

### DISCUSSION THREAD

**Discussion by michaelpolakatwork** [polatory/polatory#141] [General]
Hi.  I am testing out the sdf_surface.exe tool by taking an existing Wavefront OBJ file, writing it out as xyz data to be used as input.  The tool requires data in the form x,y,z,value.  I'm not sure what value should be so I just put 1.0 for each point.  I have the mesh-bbox information by looking at the bounding box that meshlab generates so that parameter is fine.  I'm not sure what value the mesh-res should have or how to specify the --rbf parameter.  I looked at one of the papers and it just seems to mention rbf centers.  I've tried the sp3 VAR SCALE option but I don't know what to put for VAR or SCALE.  An example on how to set these would be great!

**Comment by unageek** [comment]
To reconstruct an object from a point cloud, you first need to estimate the per-point normal and save it in `X,Y,Z,NX,NY,NZ` format. Then feed it to `normals_sdf` to generate a signed distance function data. Its output format is `X,Y,Z,VAL`. Now you can use `sdf_surface` to do interpolation and surface generation. Try `--rbf bh3 1`. The `--mesh-res` parameter specifies the size of the triangles in the output mesh, so it should be 0.01 or so for a 1-meter object.
